packages/flutter_test/test/window_test.dart

```diff
@@ -3,239 +3,552 @@
+      verifyPropertyFaked<AccessibilityFeatures>(
+        tester: tester,
+        realValue: ui.window.accessibilityFeatures,
+        fakeValue: const FakeAccessibilityFeatures(),
+        propertyRetriever: () {
+          return WidgetsBinding.instance.window.accessibilityFeatures;
+        },
+        propertyFaker: (TestWidgetsFlutterBinding binding, AccessibilityFeatures fakeValue) {
+          binding.window.accessibilityFeaturesTestValue = fakeValue;
+        },
+      );
+    });
+
+    testWidgets('can fake platform brightness', (WidgetTester tester) async {
+      verifyPropertyFaked<Brightness>(
+        tester: tester,
+        realValue: Brightness.light,
+        fakeValue: Brightness.dark,
+        propertyRetriever: () {
+          return WidgetsBinding.instance.window.platformBrightness;
+        },
+        propertyFaker: (TestWidgetsFlutterBinding binding, Brightness fakeValue) {
+          binding.window.platformBrightnessTestValue = fakeValue;
+        },
+      );
+    });
+
+    testWidgets('can clear out fake properties all at once', (WidgetTester tester) async {
+      final double originalDevicePixelRatio = ui.window.devicePixelRatio;
+      final double originalTextScaleFactor = ui.window.textScaleFactor;
+      final TestWindow testWindow = retrieveTestBinding(tester).window;
+
+      // Set fake values for window properties.
+      testWindow.devicePixelRatioTestValue = 2.5;
+      testWindow.textScaleFactorTestValue = 3.0;
+
+      // Erase fake window property values.
+      testWindow.clearAllTestValues();
+
+      // Verify that the window once again reports real property values.
+      expect(WidgetsBinding.instance.window.devicePixelRatio, originalDevicePixelRatio);
+      expect(WidgetsBinding.instance.window.textScaleFactor, originalTextScaleFactor);
+    });
+
+    testWidgets('sends fake locales when WidgetsBindingObserver notifiers are called', (WidgetTester tester) async {
+      final List<Locale> defaultLocales = WidgetsBinding.instance.window.locales;
+      final TestObserver observer = TestObserver();
+      retrieveTestBinding(tester).addObserver(observer);
+      final List<Locale> expectedValue = <Locale>[const Locale('fake_language_code')];
+      retrieveTestBinding(tester).window.localesTestValue = expectedValue;
+      expect(observer.locales, equals(expectedValue));
+      retrieveTestBinding(tester).window.localesTestValue = defaultLocales;
+    });
   });
 
-  testWidgets('TestWindow can fake device pixel ratio', (WidgetTester tester) async {
-    verifyThatTestWindowCanFakeProperty<double>(
-      tester: tester,
-      realValue: ui.window.devicePixelRatio,
-      fakeValue: 2.5,
-      propertyRetriever: () {
-        return WidgetsBinding.instance.window.devicePixelRatio;
-      },
-      propertyFaker: (TestWidgetsFlutterBinding binding, double fakeValue) {
-        binding.window.devicePixelRatioTestValue = fakeValue;
-      },
-    );
+  group('TestFlutterView', () {
```

Comment:
Definitely up for moving these tests into another file. There aren't really the same dependency issues that the `window.dart` file has.

Where would I put shared utilities, because there are a couple in here?

Reply:
You can just put them in a file not ending in `_test.dart` alongside the test files.

Reply:
Done. I also brought `platform_dispatcher_test.dart` up to date with those utility methods.